src/orquestra/sdk/_base/_api/_wf_run.py

```diff
@@ -431,11 +433,73 @@ def get_logs(self) -> WorkflowLogs:
         """
         return self._runtime.get_workflow_logs(wf_run_id=self.run_id)
 
-    # TODO: ORQSDK-617 add filtering ability for the users
-    def get_tasks(self) -> t.Set[TaskRun]:
-        wf_run_model = self.get_status_model()
+    def get_tasks(
+        self,
+        state: t.Optional[t.Union[State, t.List[State]]] = None,
+        function_name: t.Optional[str] = None,
+        task_run_id: t.Optional[t.Union[str, TaskRunId]] = None,
+        task_invocation_id: t.Optional[t.Union[str, ir.TaskInvocationId]] = None,
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Let's make these keyword only args to prevent mis-filters.
```suggestion
        self,
        *,
        state: t.Optional[t.Union[State, t.List[State]]] = None,
        function_name: t.Optional[str] = None,
        task_run_id: t.Optional[t.Union[str, TaskRunId]] = None,
        task_invocation_id: t.Optional[t.Union[str, ir.TaskInvocationId]] = None,
```